### PR TITLE
Deprecate / refactor EventDispatcher's multitude of send/receive/yield/handle events methods

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
Blocked. Will test with dhtproto.

**Comment by gavin-norman-sociomantic** [comment]
>Will test with dhtproto.

Interesting. This works without any deprecation messages. dhtproto must only use the simple send/receive methods and the `RequestEventDispatcher` methods (which aren't changed by this PR).

**Comment by gavin-norman-sociomantic** [comment]
Tested with dmqproto. Compiles and tests with deprecation warnings, as expected. I tried converting a couple of `sendReceive` calls to use `nextEvent` and it seemed pretty straightforward. I feel this is the right road to go down.

Unblocked.

File: src/swarm/neo/connection/RequestOnConnBase.d [hunk 1]
```diff
@@ -1014,22 +1014,13 @@ abstract class RequestOnConnBase
...
-            if (resume_code >= 0)
-            {
-                assert(on_other_resume_code !is null,
-                       "User unexpectedy resumed the fiber");
-                on_other_resume_code(resume_code);
-            }
+            NextEventFlags flags;
+            flags |= NextEventFlags.Receive;
+            auto event = this.nextEvent(flags);
+            assert(event.active == event.active.received);
```
Comment (david-eckardt-sociomantic) [inline src/swarm/neo/connection/RequestOnConnBase.d]: - Why not `auto event = this.nextEvent(NextEventFlags.Receive);`?
- The assertion is done already when evaluating `event.received`.

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/connection/RequestOnConnBase.d]: >Why not auto event = this.nextEvent(NextEventFlags.Receive);?

Right :man_facepalming: 

>The assertion is done already when evaluating event.received.

Ah, indeed.

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/connection/RequestOnConnBase.d]: Updated.

**Comment by gavin-norman-sociomantic** [comment]
Rebased.

**Review by david-eckardt-sociomantic — APPROVED** [review]

**Comment by gavin-norman-sociomantic** [comment]
Auto-merged after David's approval.

**Pull request merged**